internal/daemon/daemon.go

```diff
@@ -433,6 +437,17 @@ func (d *Daemon) StartAPI(bootstrap bool, initConfig map[string]string, newConfi
 			return err
 		}
 
+		// Start all extension servers before post-bootstrap hook.
+		for _, extensionServer := range d.extensionServers {
```

Comment:
this loop looks identical to the one below just before the post join hook. Maybe this can become a function?

Reply:
Fixed.